arch/arm/core/fault.c

```diff
@@ -554,35 +503,18 @@ static void _ReservedException(const NANO_ESF *esf, int fault)
 {
 	ARG_UNUSED(esf);
 
-	PR_EXC("***** %s %d) *****\n",
+	PR_FAULT_INFO("***** %s %d) *****\n",
 	       fault < 16 ? "Reserved Exception (" : "Spurious interrupt (IRQ ",
 	       fault - 16);
 }
 
-/**
- *
- * @brief Dump information regarding fault (FAULT_DUMP == 2)
- *
- * Dump information regarding the fault when CONFIG_FAULT_DUMP is set to 2
- * (long form), and return the error code for the kernel to identify the fatal
- * error reason.
- *
- * eg. (precise bus error escalated to hard fault):
- *
- * Executing thread ID (thread): 0x200000dc
- * Faulting instruction address:  0x000011d3
- * ***** HARD FAULT *****
- *    Fault escalation (see below)
- * ***** BUS FAULT *****
- *   Precise data bus error
- *   Address: 0xff001234
- *
- * @return error code to identify the fatal error reason
- */
-static u32_t _FaultDump(const NANO_ESF *esf, int fault)
+/* Handler function for ARM fault conditions. */
+static u32_t _FaultHandle(const NANO_ESF *esf, int fault)
 {
 	u32_t reason = _NANO_ERR_HW_EXCEPTION;
 
+	_FaultThreadShow(esf, fault);
+
 	switch (fault) {
 	case 3:
```

Comment:
nitpick, unrelated to your patch: Are there defined constants we could use instead of just raw numbers for these switch cases?

Reply:
>nitpick, unrelated to your patch: Are there defined constants we could use instead of just raw numbers for these switch cases?

No, the ARM CMSIS do no include any defined macros for these.